ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -336,13 +349,26 @@ implSyncWithLedger mpEnv = getStatePair mpEnv (StaticLeft ()) [] [] >>= \case
   GetStatePair
 -------------------------------------------------------------------------------}
 
--- | Get the current state pair and the current ledger state loaded with all the
--- necessary 'ValuesMK'
+-- | Get the current mempool internal state and the ledger state at the tip of
+-- the chain, loaded with all the 'ValuesMK' that are needed to validate the
+-- both the transactions in the mempool as well as the transactions provided as
+-- parmeters.
 --
--- If a desired block point is given, then no ledger state is only returned if
--- that point is not on the current chain.
+-- This function allows to specify the keys in the internal state we do not want
+-- to fetch the inputs from (see @removals@ below).
+--
+-- In the 'StaticLeft' case, this function returns 'Nothing' when there are no
+-- removals and additions (the last two parameters are empty) __and__ the point
+-- of the internal state coincides with that of the ledger state at the tip of
+-- the chain. If the aforementioned conditions are met, then we already have in
+-- the ledger state all the values that we need to validate the transactions in
+-- the internal state ('isTxs').
+--
+-- In the 'StaticRight' case, if a block point is given and that point is not on
+-- the current chain, then no ledger state is returned.
 --
 -- NOTE: The ledger state is not necessarily the anchor of the 'InternalState'!
+-- FIXME: What is the anchor of the internal state?
```

Comment:
I am pretty sure that `anchor` means the ledger state referenced by `isSlotNo` and `isTip` (not sure about the naming of that last one, I mean the one that has the Hash reference for the ledger state on top of which the txs were applied).